### PR TITLE
orocksdb 0.1.0

### PR DESCRIPTION
**Pull request by toolslive**

### CODE REVIEW

**Comment by domsj** [comment]
Regarding the travis result:
https://travis-ci.org/ocaml/opam-repository/jobs/71748029
Supposedly that build succeeded, while in reality it failed to install due to ppx_deriving install that fails.

(Which reminded me that orocksdb doesn't really need ppx_deriving, so removed that from the META file, can be picked up in a next version...)

**Comment by toolslive** [comment]
will adjust md5sum, add some external dependencies and update here.

**Comment by toolslive** [comment]
regarding the `sudo` step,
we could install the rocksdb `.so` in stublibs with something like this in the Makefile
```Makefile
OCAML_SO_DIR ?= ${OPAMROOT}/`opam switch show`/lib/stublibs
build:
	ocamlbuild -use-ocamlfind -lflags -ccopt,-L$(OCAML_SO_DIR) -lflags -cclib,-lrocksdb rocks.inferred.mli rocks.cma rocks.cmxa rocks.cmxs
```

but first, we need to convince travis to compile it.

File: packages/orocksdb/orocksdb.0.1.0/opam [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+]
+install: [make "install"]
+remove: ["ocamlfind" "remove" "rocks"]
+depends: [
+  "ocamlfind" {build}
+  "ctypes" { >= "0.4.0" }
+  "ctypes-foreign" { >= "0.4.0"}
+]
+
+available: [ "ocaml-version" >= "4.02.1" ]
```
Comment (yallop) [inline packages/orocksdb/orocksdb.0.1.0/opam]: This should be

```
available: [ ocaml-version >= "4.02.1" ]
```

(i.e. without quotes around `ocaml-version`.)

**Comment by yallop** [comment]
If there are no Debian and Ubuntu packages available then you can use a `["source" "linux"]` depext to point to an installation script.  See the `qfs.0.4` `opam` file for an example:

https://github.com/ocaml/opam-repository/blob/e0b71a27/packages/qfs/qfs.0.4/opam#L27

File: packages/orocksdb/orocksdb.0.1.0/opam (hunk 1 above)
Comment (toolslive) [inline packages/orocksdb/orocksdb.0.1.0/opam]: opam lint was happy ....

Comment (yallop) [inline packages/orocksdb/orocksdb.0.1.0/opam]: Yes, it'd be useful if opam lint warned about this.

**Comment by toolslive** [comment]
There are multiple problems:
- there's not rocksdb debian package, so we must compile it ourselves. (that's ok, we have a script)
- we need a g++-4.8 to do so. There's a debian package for that, and we register it in depexts, but travis does not listen.

File: packages/orocksdb/orocksdb.0.1.0/opam [hunk 2]
```diff
@@ -0,0 +1,31 @@
+opam-version: "1.2"
+name: "orocksdb"
+version: "0.1.0"
+maintainer: "Jan Doms <jan.doms@openvstorage.com>"
+authors: "Jan Doms <jan.doms@openvstorage.com>"
+homepage: "https://github.com/domsj/orocksdb"
+bug-reports: "https://github.com/domsj/orocksdb"
+license: "MIT"
+dev-repo: "https://github.com/domsj/orocksdb"
```
Comment (yallop) [inline packages/orocksdb/orocksdb.0.1.0/opam]: This needs a `.git` suffix:
```
dev-repo: "https://github.com/domsj/orocksdb.git"
```

**Comment by toolslive** [comment]
in fact travis does install the listed packages:
https://travis-ci.org/ocaml/opam-repository/jobs/71817983#L469
but the later bails out during compilation of rocksdb:
https://travis-ci.org/ocaml/opam-repository/jobs/71817983#L616

So I suspect we get a different c++ compiler
How do I get the contents of these ?

https://travis-ci.org/ocaml/opam-repository/jobs/71817983#L615
https://travis-ci.org/ocaml/opam-repository/jobs/71817983#L603

**Comment by samoht** [comment]
> How do I get the contents of these ?

I think you can set `OPAMERRLOGLEN` to an higher values in `.travis.yml` (default is 12).

**Comment by toolslive** [comment]
https://travis-ci.org/ocaml/opam-repository/jobs/72256955#L572

The install script still receives the wrong C++ compiler.

**Comment by yallop** [comment]
It appears that both g++ 4.6 and g++ 4.8 are installed, and /usr/bin/g++ is a symlink to g++-4.6.  Does it help if you specify the c++ compiler explicitly in the build instructions, e.g. by changing this line

```bash
make shared_lib
```

([here](https://gist.github.com/toolslive/56b9fae64b27d3e00971/#file-gistfile1-txt-L12)) to this

```bash
CXX=g++-4.8 make shared_lib
```

?

**Comment by toolslive** [comment]
it passes, except for opam 1.1. Didn't I state I have  opam 1.2 ?
https://github.com/ocaml/opam-repository/pull/4506/files#diff-83bec4475e7e506b603823f67a373b11R1

**Comment by yallop** [comment]
The `version: 1.2` field makes it possible to automatically rewrite some fields into a form that's accepted by earlier OPAM versions.  Perhaps there's a difference in depext support between the OPAM versions that's not handled by the rewrite logic.

If you want to restrict the package to OPAM versions 1.2 and above you can change the `available` field to

```
available: [ocaml-version >= "4.02.1" & opam-version >= "1.2"]
```

**Comment by toolslive** [comment]
travis is happy now, but people having a different c++ compiler (for example g++-4.9 or clang++) will not be so happy.  I guess that's for a future version?

**Comment by yallop** [comment]
Right.  A configure step to detect a suitable C++ compiler would be useful, but either automatic installation (i.e. `opam depext -i orocksdb`) using g++-4.8 or manual installation of dependencies should work for the moment.

**Comment by toolslive** [comment]
great! thx for all the help on getting this in.

**Pull request merged**